gunrock/app/mf/mf_enactor.cuh

```diff
@@ -7,180 +7,331 @@
+		if (height[src] <= height[dest])
+		{
+	//	    printf("relabel src %d, dest %d, H[%d]=%d, -> %d\n",\
+			src, dest, src, height[src], height[dest]+1);
+		    height[src] = height[dest] + 1;
+		    active[0] = 1;
+		    return true;
+		}
+	    }
+	    return false;
+	};
+	
+//	GUARD_CU(excess.ForAll(
+//	    [] __host__ __device__ (ValueT *excess_, const SizeT &v){
+//	      printf("excess_[%d] = %lf\n", v, excess_[v]);
+//	    }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+//
+//	GUARD_CU(height.ForAll(
+//	    [] __host__ __device__ (VertexT *h, const SizeT &v){
+//	      printf("height[%d] = %d\n", v, h[v]);
+//	    }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+//
+//	GUARD_CU(flow.ForAll(
+//	    [] __host__ __device__ (ValueT *f, const SizeT &v){
+//	      printf("flow[%d] = %lf\n", v, f[v]);
+//	    }, graph.edges, util::DEVICE, oprtr_parameters.stream));
+
+	
+	oprtr_parameters.advance_mode = "ALL_EDGES";
+	GUARD_CU(active.ForAll(
+	    [] __host__ __device__ (SizeT *a, const SizeT &v){
+	      a[v] = 0;
+	    }, 1, util::DEVICE, oprtr_parameters.stream));
+	
+	// ADVANCE_PUSH_OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_push_op));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+	
+	GUARD_CU(lowest_neighbor.ForAll(
+          [] __host__ __device__ (VertexT *el, const SizeT &v){
+	    el[v] = util::PreDefinedValues<VertexT>::InvalidValue;
+          }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+          "cudaStreamSynchronize failed");
+
+
+	// ADVANCE_FIND_LOWEST_OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_find_lowest_op)); 
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+      
+//	GUARD_CU(lowest_neighbor.ForAll(
+//          [] __host__ __device__ (VertexT *el, const SizeT &v){
+//            printf("lowest_neighbor[%d] = %d\n", v, el[v]);
+//          }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+          "cudaStreamSynchronize failed");
+
+	// ADVANCE RELABEL OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_relabel_op));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+	
+//	GUARD_CU(active.ForAll(
+//	    [] __host__ __device__ (SizeT *a, const SizeT &v){
+//	      if (a[v]){
+//		printf("there are");
+//	      }else{
+//		printf("there are not");
+//	      }
+//	      printf(" active nodes\n");
+//	    }, 1, util::DEVICE, oprtr_parameters.stream));
+
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
```

Comment:
You don't need this `cudaStreamSynchronize`, there's one already in place in line 238.

Reply:
removed